force-app/recipes/classes/mocking/WhenCalledWithMatchingThrowsAndReturns.cls

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright (c) 2022, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+@isTest
+private class WhenCalledWithMatchingThrowsAndReturns {
+  @isTest
+  static void recipe() {
+    // Arrange
+    final Mock deliveryServiceMock = Mock.forType(DeliveryService.class);
+    final MethodSpy planDeliverySpy = deliveryServiceMock.spyOn('planDelivery');
+    final Bakery sut = new Bakery((DeliveryService) deliveryServiceMock.stub);
+    planDeliverySpy.returns(Date.today().addDays(2));
+    planDeliverySpy.throwsException(new RecipeException());
```

Comment:
fix: remove this arrange step so that the test is more clear

Reply:
I think it is useful, because it helps understand the rule execution order
I change the whenCalledWith thenThrow to match something else and change the "Act" to match this configuration